src/main/java/com/duke/Duke.java

```diff
@@ -0,0 +1,241 @@
+package com.duke;
+
+import com.duke.command.Command;
+import com.duke.task.TaskList;
+import com.duke.util.DukeException;
+import com.duke.util.Parser;
+import com.duke.util.Storage;
+import com.duke.util.Ui;
+
+import java.io.FileNotFoundException;
+
+/**
+ * Driver class of the Duke program. To start a new session,
+ * instantiate a new <code>Duke</code> object with the data file path and call run();
+ */
+public class Duke {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Instantiate a new <code>Duke</code> object with the data file path
+     * that stores the task information.
+     *
+     * @param filePath the file path of the file that stores the task information.
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (FileNotFoundException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Starts a Duke session.
+     */
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
+        while (!isExit) {
+            try {
+                String fullCommand = ui.readCommand();
+                ui.showLine(); // show the divider line ("_______")
+                Command c = Parser.parse(fullCommand);
+                c.execute(tasks, ui, storage);
+                isExit = c.isExit;
+            } catch (DukeException e) {
+                ui.showError(e.getMessage());
+            } finally {
+                ui.showLine();
+            }
+        }
+    }
+
+
+    public static void main(String[] args) {
+        new Duke("../duke/data/duke.txt").run();
+
+    }
+}
+
+
+
+
+
+
+
+
+
+
+/*
```

Comment:
You can remove these code - they will still be available on github from your previous commits (if you ever need them again)